sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1076,6 +1076,16 @@ class SessionCatalog(
       externalCatalog.functionExists(db, name.funcName)
   }
 
+  def buildinFunctionExists(name: FunctionIdentifier): Boolean = {
+    functionRegistry.functionExists(name)
+  }
+
+  def externalFunctionExists(name: FunctionIdentifier): Boolean = {
+    val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
+    requireDbExists(db)
+    externalCatalog.functionExists(db, name.funcName)
+  }
+
```

Comment:
To avoid code duplication, you should modify the `functionExists` like:
```
def functionExists(name: FunctionIdentifier): Boolean = {
  buildinFunctionExists(name) || externalFunctionExists(name)
}
```

Reply:
ok, I will change.